staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -204,6 +213,10 @@ func (f *fancyResponseWriterDelegator) Flush() {
 func (f *fancyResponseWriterDelegator) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	// fake a response status before protocol switch happens
 	f.processCode(http.StatusSwitchingProtocols)
+
+	// TODO(caoshufeng): we can't call WriteHeader before Hijack(), because WriteHeader may be called later:
```

Comment:
The latter, but @CaoShuFeng says it doesn't work, b/c in case of proxying request the headers are sent by the kubelet and not the apiserver